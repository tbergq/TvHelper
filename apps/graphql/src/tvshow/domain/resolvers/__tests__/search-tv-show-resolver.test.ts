import searchTvshowResolver from '../search-tv-show-resolver';

const setup = () => {
  const args = {
    query: 'game of thrones',
  };
  const search = jest.fn();
  const tvshowService = {
    search,
  };
  const context: any = {
    tvshowService,
  };
  const resolve = () => searchTvshowResolver(null, args, context);
  return {
    resolve,
    search,
  };
};

it('resolves correctly', async () => {
  const { resolve, search } = setup();
  search.mockResolvedValue([
    {
      id: 1,
      name: 'Game of Thrones',
      status: 'running',
      premiered: '2011-04-17',
      image: { medium: 'mock_medium', original: 'mock_original' },
      rating: 5,
      network: { id: 5, name: 'HBO' },
    },
  ]);
  const result = await resolve();
  expect(result).toMatchSnapshot();
  expect(search).toHaveBeenCalledWith('game of thrones');
});
