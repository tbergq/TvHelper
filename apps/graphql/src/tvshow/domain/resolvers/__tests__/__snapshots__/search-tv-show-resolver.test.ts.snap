// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolves correctly 1`] = `
Object {
  "edges": Array [
    Object {
      "cursor": "YXJyYXljb25uZWN0aW9uOjA=",
      "node": Object {
        "id": 1,
        "image": Object {
          "medium": "mock_medium",
          "original": "mock_original",
        },
        "name": "Game of Thrones",
        "network": Object {
          "id": 5,
          "name": "HBO",
        },
        "premiered": "2011-04-17",
        "rating": 5,
        "status": "running",
      },
    },
  ],
  "pageInfo": Object {
    "endCursor": "YXJyYXljb25uZWN0aW9uOjA=",
    "hasNextPage": false,
    "hasPreviousPage": false,
    "startCursor": "YXJyYXljb25uZWN0aW9uOjA=",
  },
}
`;
